### PR TITLE
Decrypt models

### PR DESCRIPTION
**Pull request by voneiden**

### CODE REVIEW

File: application_form/models/application.py [hunk 1]
```diff
@@ -85,7 +73,7 @@ class ApplicationApartment(models.Model):
         Application, on_delete=models.CASCADE, related_name="application_apartments"
     )
     apartment_uuid = models.UUIDField(verbose_name=_("apartment uuid"))
-    priority_number = IntegerPGPPublicKeyField(_("priority number"))
+    priority_number = models.IntegerField(_("priority number"))
 
     class Meta:
         unique_together = [("application", "priority_number")]
```
Comment (voneiden) [inline application_form/models/application.py:79]: I wonder if this actually ever worked before. Discovered some failing tests related to this

```
=========================== short test summary info ============================
ERROR application_form/tests/test_export_service.py::test_export_applicants - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(116, 1) already exists.
ERROR application_form/tests/test_export_service.py::test_export_applicants_and_secondary_applicants - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(117, 5) already exists.
ERROR application_form/tests/test_export_service.py::test_csv_output - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(160, 2) already exists.
ERROR application_form/tests/test_export_service.py::test_csv_non_ascii_characters - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(161, 3) already exists.
ERROR application_form/tests/test_export_service.py::test_write_csv_file - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(162, 3) already exists.
FAILED apartment/tests/api/test_views.py::test_project_detail_apartment_reservations_multiple_winning - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(35, 1) already exists.
FAILED application_form/tests/api/test_project_reservation_view.py::test_list_project_reservations_get - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(308, 4) already exists.
FAILED application_form/tests/api/test_project_reservation_view.py::test_list_project_reservations_get_without_lottery_data - django.db.utils.IntegrityError: duplicate key value violates unique constraint "application_form_applica_application_id_priority__89530de7_uniq"
DETAIL:  Key (application_id, priority_number)=(309, 3) already exists.
FAILED invoicing/tests/test_migrations.py::test_foo - django.db.utils.IntegrityError: null value in column "type" violates not-null constraint
DETAIL:  Failing row contains (49, 2024-08-18 19:10:02.196468+00, null, null, , null, TEMP-3b3e8452-6d72-4ea5-9178-5555c04892b1, null, 730000001, null, null, 2024-08-18 19:10:02.196472+00, ).
====== 4 failed, 419 passed, 4449 warnings, 5 errors in 150.73s (0:02:30) ======
```

File: apartment_application_service/utils.py [hunk 2]
```diff
@@ -25,3 +28,54 @@ def is_module_available(module_name):
...
+    """
+    if reverse:
+        to_fields = fields
+        from_fields = [f"{field}_plain" for field in fields]
+    else:
+        to_fields = [f"{field}_plain" for field in fields]
+        from_fields = fields
+
+    instances = model.objects.all()
+    for instance in instances:
```
Comment (voneiden) [inline apartment_application_service/utils.py:45]: Use `.iterator()` ?

File: apartment_application_service/utils.py [hunk 3]
```diff
@@ -25,3 +28,54 @@ def is_module_available(module_name):
...
+    Helper function for decrypt migrations
+    """
+    if reverse:
+        to_fields = fields
+        from_fields = [f"{field}_plain" for field in fields]
+    else:
+        to_fields = [f"{field}_plain" for field in fields]
+        from_fields = fields
+
+    instances = model.objects.all()
```
Comment (voneiden) [inline apartment_application_service/utils.py:44]: How about also using `.only(fields)`

File: application_form/tests/test_migrations.py [hunk 4]
```diff
@@ -233,3 +239,261 @@ def assert_field_type(model, field_name, expected_type):
...
+        "is_primary_applicant": models.BooleanField,
+        "last_name": models.CharField,
+        "phone_number": models.CharField,
+        "postal_code": models.CharField,
+        "street_address": models.CharField,
+        "date_of_birth": pgcrypto.fields.DatePGPPublicKeyField,  # bonus check
+        "ssn_suffix": pgcrypto.fields.CharPGPPublicKeyField,  # bonus check
+    }
+
+    application_apt_encrytped_field_map = {
```
Comment (tuomas777) [inline application_form/tests/test_migrations.py:313]: `encrytped` !

File: application_form/tests/test_migrations.py [hunk 5]
```diff
@@ -233,3 +239,261 @@ def assert_field_type(model, field_name, expected_type):
...
+        "priority_number": IntegerPGPPublicKeyField,
+    }
+
+    application_apt_decrytped_field_map = {
```
Comment (tuomas777) [inline application_form/tests/test_migrations.py:317]: `decrytped` !!!

**Review by tuomas777 — APPROVED** [review]
Looks very good to me 🏅

**Pull request merged**